ext/spl/spl_fixedarray.c

```diff
@@ -358,6 +358,20 @@ static zval *spl_fixedarray_object_read_dimension(zval *object, zval *offset, in
 
 	intern = Z_SPLFIXEDARRAY_P(object);
 
+	if (type == BP_VAR_IS) {
```

Comment:
So what I had in mind here is something like this:

```php
<?php
class MyArray extends SplFixedArray {
    public function offsetGet($key) {
        return "prefix_" . parent::offsetGet($key);
    }
}

$arr = new MyArray(1);
var_dump(isset($arr[0]) ? $arr[0] : 42); // Returns 42
var_dump($arr[0] ?? 42); // According to your patch (I think) this will throw rather than returning 42
```

Due to a typo only extending offsetGet didn't actually work, just fixed this in https://github.com/php/php-src/commit/1967950dc33b87a5b92c33875885cfb0b5b5bffc.